test/integration/smoke/test_deploy_vm_extra_config_data.py

```diff
@@ -0,0 +1,522 @@
+                xml_as_str = ''.join(xml_res)
+
+                extraconfig_decoded_xml = '<config>' + urllib.unquote(extraconfig) + '</config>'
+
+                # Root XML Elements
+                parser = etree.XMLParser(remove_blank_text=True)
+                domain_xml_root = ET.fromstring(xml_as_str, parser=parser)
+                decoded_xml_root = ET.fromstring(extraconfig_decoded_xml, parser=parser)
+                for child in decoded_xml_root:
+                    find_element_in_domain_xml = domain_xml_root.find(child.tag)
+
+                    # Fail if extra config is not found in domain xml
+                    self.assertNotEquals(
+                        0,
+                        len(find_element_in_domain_xml),
+                        'Element tag from extra config not added to VM'
+                    )
+
+                    # Compare found XML node with extra config node
+                    is_a_match = self.elements_equal(child, find_element_in_domain_xml)
+                    self.assertEquals(
+                        True,
+                        is_a_match,
+                        'The element from tags from extra config do not match with those found in domain xml'
+                    )
+            finally:
+                self.destroy_vm(response.id)
+                self.add_global_config(name, "")
+
+    @attr(tags=["devcloud", "advanced", "advancedns", "smoke", "basic", "sg"], required_hardware="true")
+    def test_03_update_vm__with_extraconfig_kvm(self):
+        '''
+        Test that extra config is added on KVM hosts
+        '''
+        if self.hypervisor.lower() != 'kvm':
+            raise self.skipTest("Skipping test case for non-kvm hypervisor")
+
+        name = 'allow.additional.vm.configuration.list.kvm'
+        value = 'memoryBacking, hugepages'
+
+        add_config_response = self.add_global_config(name, value)
+
+        if add_config_response.name:
+            try:
+                '''
+                The following extraconfig is required for enabling hugepages on kvm
+                <memoryBacking>
+                    <hugepages/>
+                </memoryBacking>
+                url encoded extra_config = '%3CmemoryBacking%3E%0D%0A++%3Chugepages%2F%3E%0D%0A%3C%2FmemoryBacking%3E'
+                '''
+                extraconfig = "%3CmemoryBacking%3E%0D%0A++%3Chugepages%2F%3E%0D%0A%3C%2FmemoryBacking%3E"
+
+                response = self.deploy_vm('kvm')
+                vm_id = response.id
+
+                '''
+                For updateVirtualMachineCmd, the VM must be stopped and restarted for changes to take effect
+                '''
+                self.stop_vm(vm_id)
+                self.update_vm(vm_id, extraconfig)
+                start_resp = self.start_vm(vm_id)
+
+                host_id = start_resp.hostid
+                host = list_hosts(
+                    self.apiclient,
+                    id=host_id,
+                    hypervisor='kvm')
```

Comment:
This shouldn't be 'kvm' hardcoded, but instead use 'testClient.getHypervisorInfo()' to set a class var

Reply:
Done, now using `self.hypervisor.lower()`